server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1106,7 +1106,7 @@ private ReplicationGroup calculateReplicationGroup() {
         } else {
             newVersion = replicationGroup.getVersion() + 1;
         }
-        assert indexSettings.isRemoteTranslogStoreEnabled()
+        assert indexSettings.isAssignedOnRemoteNode()
```

Comment:
Can we keep the existing `isRemoteTranslogStoreEnabled` assertion here? 

I see why this is required since during local recovery the `replicationGroup` object is never instantiated. During a usual primary failover, the `initiateTracking` method invocation would ideally instantiate the object:

https://github.com/opensearch-project/OpenSearch/blob/e617da375740065a3878cea093ac8eadaf2a821a/server/src/main/java/org/opensearch/indices/recovery/LocalStorePeerRecoverySourceHandler.java#L191-L197

Instead, can we add the change the assertion to something like this?

```
assert newVersion == 0 || indexSettings.isRemoteTranslogStoreEnabled()
            // Handle migration cases. Ignore assertion if any of the shard copies in the replication group is assigned to a remote node
            || replicationGroup.getReplicationTargets()
                    .stream()
                    .anyMatch(shardRouting -> isShardOnRemoteEnabledNode.apply(shardRouting.currentNodeId()))
            || checkpoints.entrySet().stream().filter(e -> e.getValue().tracked).allMatch(e -> e.getValue().replicated)
            : "In absence of remote translog store, all tracked shards must have replication mode as LOGICAL_REPLICATION";
```

Reply:
@gbbafna, @linuxpi, @shourya035 as per Shourya's comment above, I have added back the `isRemoteTranslogStoreEnabled()` check and added the new `newVersion == 0` check